test/rules/std_L048_test.py

```diff
@@ -0,0 +1,66 @@
+"""Tests the python routines within L048."""
+import pytest
+
+import sqlfluff
+
+
+def test__rules__std_L048() -> None:
+    """L0048 is not raised for quoted literals surrounded by a single whitespace."""
+    sql = "SELECT a + 'b' + 'c' FROM tbl;"
+    result = sqlfluff.lint(sql)
+
+    assert len(result) == 1
+
+    results_l048 = [r for r in result if r["code"] == "L048"]
+    assert not len(results_l048)
+
+
+def test__rules__std_L048_raised() -> None:
+    """L0048 is raised for quoted literals not surrounded by a single whitespace."""
+    sql = "SELECT a +'b'+'c' FROM tbl;"
+    result = sqlfluff.lint(sql)
+
+    assert len(result) == 7
+
+    results_l048 = [r for r in result if r["code"] == "L048"]
+    assert len(results_l048) == 3
+    assert results_l048[0]["description"] == "Missing whitespace before 'b'"
+    assert results_l048[1]["description"] == "Missing whitespace after 'b'"
+    assert results_l048[2]["description"] == "Missing whitespace before 'c'"
+
+
+def test__rules__std_L048_unicode_tsql() -> None:
+    """L0048 is not raised for unicode quoted literals when T-SQL."""
+    sql = "SELECT a + N'b' + N'c' FROM tbl;"
+    result = sqlfluff.lint(sql, dialect="tsql")
+
+    assert len(result) == 1
+
+    results_l048 = [r for r in result if r["code"] == "L048"]
+    assert not len(results_l048)
+
+
+@pytest.mark.parametrize("dialect", ["ansi", "mysql", "postgres", "sqlite", "bigquery"])
+def test__rules__std_L048_unicode_other_dialects(dialect: str) -> None:
+    """L0048 is raised for unicode quoted literals when some other dialects."""
+    sql = "SELECT a + N'b' + N'c' FROM tbl;"
+    result = sqlfluff.lint(sql, dialect=dialect)
+
+    assert len(result) == 3
+
+    results_l048 = [r for r in result if r["code"] == "L048"]
+    assert len(results_l048) == 2
+    assert results_l048[0]["description"] == "Missing whitespace before 'b'"
+    assert results_l048[1]["description"] == "Missing whitespace before 'c'"
```

Comment:
Covered by `test_fails_tsql_unicode_singe_quote` in YML.

Don't really need to test every dialect (and unsustainable as we add more dialects).

Also it's not valid SQL in any of the other AFAIK so even though it throws L048 error it's still not vali so question value of this case on top of the `test_fails_tsql_unicode_singe_quote` you already added in YML.